src/compiler/checker.ts

```diff
@@ -10702,8 +10779,24 @@ namespace ts {
                 }
 
                 let derived = getTargetSymbol(getPropertyOfObjectType(type, base.name));
-                if (derived) {
-                    let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+                let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+
+                Debug.assert(!!derived, "derived should point to something, even if it is the base class' declaration.");

```

Comment:
Seems like you should just be able to pass in derived without `!!`. I feel like Debug.assert should take an any